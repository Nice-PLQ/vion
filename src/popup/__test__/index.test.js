import { mount } from '@vue/test-utils';
import { next } from '../../../test';
import Popup from '../popup.vue';

describe('Popup', () => {
  const wrapper = mount(Popup, {
    slots: {
      default: '<div style="height: 200px; background: #fff;"></div>'
    },
    props: {
      teleportDisabled: true,
    },
  });
  test('渲染', async () => {
    wrapper.setProps({ visible: true });
    await next();
    expect(wrapper.find('.v-popup__mask').isVisible()).toBe(true);
    expect(wrapper.find('.v-popup__content').isVisible()).toBe(true);
    expect(wrapper.find('.v-popup').element).toMatchSnapshot();
  });

  test('close事件触发', () => {
    wrapper.find('.v-popup__mask').trigger('click');
    expect(wrapper.emitted('close')).toBeTruthy();
  });
  test('测试关闭', async () => {
    wrapper.setProps({ visible: false });
    await next();
    expect(document.body.classList.length).toBe(0);
  });
  test('测试contentTransitionName', async () => {
    expect(wrapper.vm.contentTransitionName).toBe('slide-bottom');
    await wrapper.setProps({
      transitionName: 'transition'
    });
    expect(wrapper.vm.contentTransitionName).toBe('transition');
    await wrapper.setProps({
      transitionName: '',
      position: 'center'
    });
    expect(wrapper.vm.contentTransitionName).toBe('fade-zoom');
  });
});
