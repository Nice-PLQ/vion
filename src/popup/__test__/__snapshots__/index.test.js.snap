// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup 渲染 1`] = `
<div
  class="v-popup"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      aria-label="关闭"
      class="v-popup__mask"
      role="button"
      style=""
    />
  </transition-stub>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="false"
  >
    <!--v-if-->
  </transition-stub>
  <transition-stub
    appear="false"
    css="true"
    name="slide-bottom"
    persisted="true"
  >
    <div
      class="v-popup__content v-popup__content_bottom"
      style=""
    >
      
      <div
        style="height: 200px; background: rgb(255, 255, 255);"
      />
      
    </div>
  </transition-stub>
</div>
`;
